lib/Cron/Workers/SimpleWorker.php

```diff
@@ -97,12 +101,34 @@ function processTask(ScheduledTask $task) {
     // abort if execution limit is reached
     CronHelper::enforceExecutionLimit($this->timer);
 
-    if ($this->processTaskStrategy($task)) {
+    if (!static::SUPPORT_MULTIPLE_INSTANCES) {
+      if ($this->isInProgress($task)) {
+        return false;
+      }
+      if ($this->rescheduleOutdated($task)) {
+        return false;
+      }
+      $this->startProgress($task);
+    }
+
+    try {
+      $completed = $this->processTaskStrategy($task);
+    } catch (\Exception $e) {
+      if (!static::SUPPORT_MULTIPLE_INSTANCES) {
```

Comment:
Just an idea. I'm thinking about moving $this->startProgress($task) and $this->endProgress($task) out of the `if (!static::SUPPORT_MULTIPLE_INSTANCES)` block so that we start storing this for all workers. I think it might be useful information for all workers. We would be able to see in DB which task is now actually being processed and could be useful during debugging.

Reply:
Sounds reasonable. Made a change, we now track progress for all workers.

Reply:
Regarding this change, while it may be handy to set this for all workers, at the time some workers may use the `meta` field in their own way which may conflict with an array having an `in_progress` key. The failed test shows that. I can fix the test, but there is still risk of introducing bugs in other workers. The most clean way to avoid such issues would be to make `in_progress` a separate column. 

We decided to roll back this change and refactor later.